homeassistant/components/envisalink/alarm_control_panel.py

```diff
@@ -173,6 +176,12 @@ def state(self):
         """Alarm trigger command. Will be used to trigger a panic alarm."""
         self.hass.data[DATA_EVL].panic_alarm(self._panic_type)
 
+    async def async_alarm_arm_night(self, code=None):
```

Comment:
This is not correct. This is doing I/O inside the event loop. You need to change the method name to the sync version. All other methods should be updated too.

```suggestion
    def alarm_arm_night(self, code=None):
```

Reply:
Because of how asyncio works, doing blocking I/O in the event loop will halt the whole system until the call is done.

Reply:
The library is using the asyncio transport and protocol based api which is using callbacks.

https://github.com/Cinntax/pyenvisalink/blob/0f57cc2f418d86e1b875ef9021c2adb1318fb45c/pyenvisalink/envisalink_base_client.py#L67-L70

So it's asyncio safe.